### PR TITLE
Added available scope & index action placeholder

### PR DESCRIPTION
**Pull request by adrianproctor**
Added :available scope statement to game model. 
Added placeholder route, controller, and view for index action in games

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class Game < ActiveRecord::Base
   validates :name, presence: true
+  scope :available, -> {where(white_player_id == nil || black_player_id == nil)}
```
Comment (mscoutermarsh) [inline app/models/game.rb]: Try writing a test for this.

We don't have factories setup yet. So this could be a good opportunity to set that up for games/users.

To get you started, this is what the test might look like (you'd have to setup the Factory for game/user for this to work)
```Ruby
it 'returns available games' do
   available_game = create(:game, white_player: user, black_player: nil)
   expect(Game.available).to include available_game
end
```

Comment (mscoutermarsh) [inline app/models/game.rb]: My goal with this is. The way the scope is written currently won't work. Writing a test will show you :).

**Comment by tohyung85** [comment]
Correct scope syntax and test for scope available
    
    - Wrote tests for scope available
    - correct scope syntax.

File: spec/models/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    it "should be true" do
+      expect(true).to eq true
+    end
+  end
+
+  describe "available scope" do
+    it "should return available games" do
+      available_game = FactoryGirl.create(:game, white_player_id: user.id)
+      expect(Game.available).to include available_game
+    end
```
Comment (mscoutermarsh) [inline spec/models/game_spec.rb]: Love this test. Now to push this 1 step further. Try writing a test that is the opposite.

```Ruby
it "returns no games if none are available" do
end
```

**Comment by tohyung85** [comment]
Include additional test to ensure available scope return empty if all Games are full
